yilan_zhou/week_01/week1-4_bank.js

```diff
@@ -0,0 +1,96 @@
+//1/10/2015  week1-4 homeowork
+//https://gist.github.com/wofockham/dacf2da17c743afb2b17
+//js bank
+
+var bank = [  //bank array with name and account
+    {name: 'Mary',
+     balance: 100
+    },{
+     name: 'David',
+     balance: 200
+    },{
+     name: 'Jane',
+     balance:300
+    }];
+
+function sumBankBalance(thisBank) {   
+    var total = 0;
+    for (var i = 0; i<thisBank.length;i++) {
+       total += thisBank[i].balance; 
+    }
+    return "Total Bank Balance: " + total;
+}
+
+function newBankAccount(thisBank,newName,newBalance) {
+    var newAccount = {
+    name: newName,
+    balance: newBalance
+};
+   thisBank.push(newAccount);
+   return "New account holder " + newName + " is added.";
+}
+
+function chgAccountBalance(thisBank,acName,acAmountChg) {
+	var result="";
+    // positive acAmountChg is deposit, negative amount is withdraw
+    for (var i =0; i<thisBank.length; i++) {
+        if(thisBank[i].name === acName) {  //if account name is found
+        	acFound = true;
+            if ((thisBank[i].balance + acAmountChg)>=0) { //if withdraw balance <= a/c balance
+                thisBank[i].balance += acAmountChg;
+                return acName + "\'s balance has changed by " + acAmountChg + " to " + thisBank[i].balance;
+            } else {  //if withdraw amount > a/c balance
+            	return "Withdraw amount " + acAmountChg + " exceeds account balance of " + thisBank[i].balance + ", pls change amount.";
+            }
+            break;
+        }
+    }
+    if (!acFound) {  //if name is not found
+        return "Name not found";
+    }
+}
+
+function accountTransfer(thisBank, fromName,toName,amount) {
```

Comment:
Same as previous function. Too many checks inside, you can take them out and make new functions, each of them should check for 1 thing, then just call them all inside of this function